webpack/__mocks__/foremanReact/Root/Context/ForemanContext.js

```diff
@@ -0,0 +1,3 @@
+// eslint-disable-next-line import/prefer-default-export
+export const useForemanSettings = () => ({ perPage: 20 });
+export const usePaginationOptions = () => [5, 10, 20, 50];
```

Comment:
Why did you have to mock these?

Reply:
They seem to eventually [dip into ruby code](https://github.com/theforeman/foreman/blob/17d7be599c88ab953d10a396eccdc93d30a69e88/app/helpers/application_helper.rb#L524) and don't seem to cooperate in a testing environment, at least in Katello. I just figured its a small thing that is easy to mock.

Reply:
This is the error if you are curious, it is a bit obscure, but a little digging showed me its these methods that are causing it, probably due to the ruby portion not being present. If you want to recreate you can just rename this mock file:

```
  ● Can handle pagination

    Error: Uncaught [Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
    1. You might have mismatching versions of React and the renderer (such as React DOM)
    2. You might be breaking the Rules of Hooks
    3. You might have more than one copy of React in the same app
    See https://fb.me/react-invalid-hook-call for tips about how to debug and fix this problem.]

      3 | // output and traceback for actual error.
      4 | global.console.error = (error) => {
    > 5 |   throw new Error(error);
        |         ^
      6 | };
      7 | 

      at reportException (node_modules/jsdom/lib/jsdom/living/helpers/runtime-script-errors.js:66:24)
      at invokeEventListeners (node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:209:9)
      at HTMLUnknownElementImpl._dispatch (node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:119:9)
      at HTMLUnknownElementImpl.dispatchEvent (node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:82:17)
      at HTMLUnknownElementImpl.dispatchEvent (node_modules/jsdom/lib/jsdom/living/nodes/HTMLElement-impl.js:30:27)
      at HTMLUnknownElement.dispatchEvent (node_modules/jsdom/lib/jsdom/living/generated/EventTarget.js:157:21)
      at Object.invokeGuardedCallbackDev (node_modules/react-dom/cjs/react-dom.development.js:237:16)
      at invokeGuardedCallback (node_modules/react-dom/cjs/react-dom.development.js:292:31)
      at beginWork$1 (node_modules/react-dom/cjs/react-dom.development.js:23203:7)
      at performUnitOfWork (node_modules/react-dom/cjs/react-dom.development.js:22157:12)
      at CustomConsole.Object.<anonymous>.global.console.error (webpack/global_test_setup.js:5:9)
      at VirtualConsole.<anonymous> (node_modules/jsdom/lib/jsdom/virtual-console.js:29:45)
      at reportException (node_modules/jsdom/lib/jsdom/living/helpers/runtime-script-errors.js:70:28)
      at invokeEventListeners (node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:209:9)
      at HTMLUnknownElementImpl._dispatch (node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:119:9)
      at HTMLUnknownElementImpl.dispatchEvent (node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:82:17)
      at HTMLUnknownElementImpl.dispatchEvent (node_modules/jsdom/lib/jsdom/living/nodes/HTMLElement-impl.js:30:27)
      at HTMLUnknownElement.dispatchEvent (node_modules/jsdom/lib/jsdom/living/generated/EventTarget.js:157:21)
      at Object.invokeGuardedCallbackDev (node_modules/react-dom/cjs/react-dom.development.js:237:16)
      at invokeGuardedCallback (node_modules/react-dom/cjs/react-dom.development.js:292:31)
      at beginWork$1 (node_modules/react-dom/cjs/react-dom.development.js:23203:7)
      at performUnitOfWork (node_modules/react-dom/cjs/react-dom.development.js:22157:12)
      at workLoopSync (node_modules/react-dom/cjs/react-dom.development.js:22130:22)
      at performSyncWorkOnRoot (node_modules/react-dom/cjs/react-dom.development.js:21756:9)
      at scheduleUpdateOnFiber (node_modules/react-dom/cjs/react-dom.development.js:21188:7)
      at updateContainer (node_modules/react-dom/cjs/react-dom.development.js:24373:3)
      at node_modules/react-dom/cjs/react-dom.development.js:24758:7
      at unbatchedUpdates (node_modules/react-dom/cjs/react-dom.development.js:21903:12)
      at legacyRenderSubtreeIntoContainer (node_modules/react-dom/cjs/react-dom.development.js:24757:5)
      at Object.render (node_modules/react-dom/cjs/react-dom.development.js:24840:10)
      at node_modules/@testing-library/react/dist/pure.js:86:25
      at batchedUpdates$1 (node_modules/react-dom/cjs/react-dom.development.js:21856:12)
      at act (node_modules/react-dom/cjs/react-dom-test-utils.development.js:929:14)
      at render (node_modules/@testing-library/react/dist/pure.js:82:26)
      at renderWithApiRedux (webpack/test-utils/react-testing-lib-wrapper.js:36:15)
      at _callee5$ (webpack/scenes/ContentViews/__tests__/contentViewPage.test.js:121:43)
      at tryCatch (node_modules/regenerator-runtime/runtime.js:45:40)
      at Generator.invoke [as _invoke] (node_modules/regenerator-runtime/runtime.js:274:22)
      at Generator.prototype.<computed> [as next] (node_modules/regenerator-runtime/runtime.js:97:21)
      at asyncGeneratorStep (webpack/scenes/ContentViews/__tests__/contentViewPage.test.js:27:103)
      at _next (webpack/scenes/ContentViews/__tests__/contentViewPage.test.js:29:194)
      at webpack/scenes/ContentViews/__tests__/contentViewPage.test.js:29:364
      at Object.<anonymous> (webpack/scenes/ContentViews/__tests__/contentViewPage.test.js:29:97)

  ● Can handle pagination

    Error: Uncaught [Error: The above error occurred in the <TableWrapper> component:
        in TableWrapper (created by ContentViewTable)
        in ContentViewTable (created by ContentViewsPage)
        in ContentViewsPage (created by Context.Consumer)
        in withRouter(ContentViewsPage)
        in Router (created by MemoryRouter)
        in MemoryRouter
        in Provider

    Consider adding an error boundary to your tree to customize error handling behavior.
    Visit https://fb.me/react-error-boundaries to learn more about error boundaries.]

      3 | // output and traceback for actual error.
      4 | global.console.error = (error) => {
    > 5 |   throw new Error(error);
        |         ^
      6 | };
      7 | 

      at reportException (node_modules/jsdom/lib/jsdom/living/helpers/runtime-script-errors.js:66:24)
      at Timeout.callback [as _onTimeout] (node_modules/jsdom/lib/jsdom/browser/Window.js:680:7)
      at CustomConsole.Object.<anonymous>.global.console.error (webpack/global_test_setup.js:5:9)
      at VirtualConsole.<anonymous> (node_modules/jsdom/lib/jsdom/virtual-console.js:29:45)
      at reportException (node_modules/jsdom/lib/jsdom/living/helpers/runtime-script-errors.js:70:28)
      at Timeout.callback [as _onTimeout] (node_modules/jsdom/lib/jsdom/browser/Window.js:680:7)
```